src/Strategy/IfExists.php

```diff
@@ -26,30 +20,12 @@ class IfExists extends Decorator
      */
     public function __construct(Strategy $condition, $if, $else = null)
     {
-        parent::__construct($condition);
-
-        $this->if = $if;
-        $this->else = $else;
-    }
-
-    /**
-     * Resolves the stored strategy or mapping when the stored condition
-     * resolves to a non-null value, otherwise returns the stored default
-     * value.
-     *
-     * @param mixed $data
-     * @param mixed $context
-     *
-     * @return mixed
-     */
-    public function __invoke($data, $context = null)
-    {
-        if (parent::__invoke($data, $context) !== null) {
-            return $this->delegate($this->if, $data, $context);
-        }
-
-        if ($this->else !== null) {
-            return $this->delegate($this->else, $data, $context);
-        }
+        parent::__construct(
+            function ($data, $context = null) use ($condition) {
```

Comment:
A callback will always receive context when it is invoked, therefore specifying a default here is redundant.